lib/rspec/core/option_parser.rb

```diff
@@ -140,8 +142,23 @@ def parser(options)
           options[:full_backtrace] = true
         end
 
-        parser.on('-c', '--[no-]color', '--[no-]colour', 'Enable color in the output.') do |o|
-          options[:color] = o
+        parser.on('-c', '--color', '--colour', 'Deprecated: enable color in the output.') do |_o|
+          options[:color] = true
+          options[:color_mode] = :automatic
+        end
```

Comment:
Update on this: if I move processing of the `--color` flag to where the `--tty` flag is processed, it seems as though it'll be difficult or impossible to retain the backwards-compatible behavior of respecting the order of `--color` and `--no-color` flags, so that the last one wins. (As a reminder, I did _not_ add last-one-wins for the new `--force-color` flag: only one of `--force-color` and `--no-color` can be specified. This is just about backwards compatibility.)

I think backwards compatibility has a higher priority than the clarity of where it is in the source code. Maybe there is another way to make it clearer that it won't be listed in the output. A comment? A change to the text "description" passed into `parser.on`?